scripts/o3de/o3de/cmake.py

```diff
@@ -243,3 +242,120 @@ def get_enabled_gem_cmake_file(project_name: str = None,
+                             f'"{project_path}" which is required to resolve gem dependencies.')
+                return 1
+
+            logger.warning('Failed to determine the correct engine for the project at '
+                           f'"{project_path}", falling back to this engine at {engine_path}.')
+    
+    engine_json_data = manifest.get_engine_json_data(engine_path=engine_path)
+    if not engine_json_data:
+        logger.error('Failed to retrieve engine json data for the engine at '
+                     f'"{engine_path}" which is required to resolve gem dependencies.')
+        return 1
+
+    if project_path:
+        project_json_data = manifest.get_project_json_data(project_path=project_path)
+        active_gem_names = project_json_data.get('gem_names',[])
+        enabled_gems_file = cmake.get_enabled_gem_cmake_file(project_path=project_path)
+        active_gem_names.extend(cmake.get_enabled_gems(enabled_gems_file))
+    else:
+        active_gem_names = engine_json_data.get('gem_names',[])
+
+    # some gem name entries will be dictionaries - convert to a set of strings 
+    gem_names_with_optional_gems = utils.get_gem_names_set(active_gem_names, include_optional=True)
+    if not gem_names_with_optional_gems:
+        logger.info(f'No gem names were found to use as input to resolve gem dependencies.')
+        with resolved_gem_dependencies_output_path.open('w') as output:
+            output.write('')
+        return 0
+
+    all_gems_json_data = manifest.get_gems_json_data_by_name(engine_path=engine_path, 
+                                                             project_path=project_path, 
+                                                             include_manifest_gems=True, 
+                                                             include_engine_gems=True)
+
+    # First try to resolve with optional gems
+    results, errors = compatibility.resolve_gem_dependencies(gem_names_with_optional_gems, 
+                                                             all_gems_json_data, 
+                                                             engine_json_data, 
+                                                             include_optional=True)
+    if errors:
+        logger.warning('Failed to resolve dependencies with optional gems, trying without optional gems.')
+
+        # Try without optional gems
+        gem_names_without_optional = utils.get_gem_names_set(active_gem_names, include_optional=False)
+        results, errors = compatibility.resolve_gem_dependencies(gem_names_without_optional, 
+                                                                 all_gems_json_data, 
+                                                                 engine_json_data,
+                                                                 include_optional=False)
+
+    if errors:
+        logger.error(f'Failed to resolve dependencies:\n '+ 
+                    "\n  ".join(errors))
```

Comment:
Why is there a single space after the first newline.
But too spaces after every other newlines.
The output would like something as follows
```
Failed to resolve dependencies
 error 1
  error 2
  error 3

Reply:
looks like a bug to me!

Reply:
In general, if you are using a logger, you wouldn't want to use anything multi-line. logger can be registered with formatters that may append things like log level, timestamp, etc, which will throw off the identing anyways.